backend/onedrive/onedrive.go

```diff
@@ -1503,7 +1503,78 @@ func (f *Fs) PublicLink(ctx context.Context, remote string, expire fs.Duration,
 		fmt.Println(err)
 		return "", err
 	}
-	return result.Link.WebURL, nil
+
+	url := result.Link.WebURL
+
+	// Convert share link to direct download link if target is not a folder
+	cnvFailMsg := "Don't know how to convert share link to direct link - returning the link as is"
+	if info.Folder != nil {
+		fs.Logf(f, cnvFailMsg)
+		return url, nil
+	}
+
+	segments := strings.Split(url, "/")
+	switch f.driveType {
+	case driveTypePersonal:
+		/*
```

Comment:
```suggestion
		// Method: https://stackoverflow.com/questions/37951114/direct-download-link-to-onedrive-file
```

In Go, `//` line comments are [the norm](https://golang.org/doc/effective_go#commentary). And this can also be a one-line comment. (also applies to others)

Reply:
KK